pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -66,23 +79,59 @@ func ApplyMonitoringConsole(client splcommon.ControllerClient, cr splcommon.Meta
 		return err
 	}
 
-	statefulsetMC, err := getMonitoringConsoleStatefulSet(client, cr, &spec, SplunkMonitoringConsole, secretName)
+	//by default set multisite to false
+	multiSite := "false"
+
+	//get cluster info from cluster master
+	if cr.GetObjectKind().GroupVersionKind().Kind == "ClusterMaster" && !spec.Mock {
+		mgr := monitoringConsolePodManager{cr: &cr, spec: &spec, secrets: secrets, newSplunkClient: splclient.NewSplunkClient}
```

Comment:
We don't need this as we could stub out getClusterMasterClient() as a function outside of the IndexerClusterPodManager interface. The function would look like:

func (splcommon.MetaObject) getClusterMasterClient() *splclient.SplunkClient 

We could place it in a common folder which can be accessible to IDXC, CM as well as MC.

Reply:
Sure, can we take this in future commits?